plugins/CoreHome/vue/src/Matomo/Matomo.ts

```diff
@@ -6,25 +6,111 @@
  */
 
 import Periods from '../Periods/Periods';
-
-let originalTitle: string;
+import { translate } from '../translate';
 
 const { piwik, broadcast, piwikHelper } = window;
 
+type ComparisonsStoreLike = {
+  getSegmentComparisons: () => Array<{params: { segment: string }, title: string, index: number}>;
+};
+
+type ReportingMenuStoreLike = {
+  findSubcategory: (categoryId: string, subcategoryId: string) => {
+    category?: { name?: string };
+    subcategory?: { name?: string };
+  };
+  fetchMenuItems: () => Promise<unknown>;
+};
+
 piwik.helper = piwikHelper;
 piwik.broadcast = broadcast;
 
-piwik.updateDateInTitle = function updateDateInTitle(date: string, period: string) {
-  if (!$('.top_controls #periodString').length) {
-    return;
+async function getReportingMenuStore(): Promise<ReportingMenuStoreLike|undefined> {
+  const coreHome = (window as unknown as {
```

Comment:
I don’t know this syntax. Is the `unknown` needed?

```suggestion
  const coreHome = (window as {
```

Reply:
That was a way to get around typescript strict type checking where it needs to have something, but your suggestion below is far better 👍